faiss/gpu/impl/IVFAppend.cu

```diff
@@ -125,6 +125,70 @@ void runIVFIndicesAppend(
     }
 }
 
+// Remove indices for vectors being removed from the IVF indices lists
+__global__ void ivfIndicesRemove(
+        Tensor<int, 1, true> listIds,
+        Tensor<int, 1, true> listOffset,
+        Tensor<int, 1, true> listReplaceOffset,
+        Tensor<Index::idx_t, 1, true> listIndicesReplaceOffset,
+        IndicesOptions opt,
+        void** listIndices) {
+    int vec = blockIdx.x * blockDim.x + threadIdx.x;
+
+    if (vec >= listIds.getSize(0)) {
+        return;
+    }
+
+    int listId = listIds[vec];
+    int offset = listOffset[vec];
+    int replaceOffset = listReplaceOffset[vec];
+
+    if (listId == -1 || offset == -1 || replaceOffset == -1) {
+        return;
+    }
+
+    if (opt == INDICES_32_BIT) {
+        ((int*)listIndices[listId])[offset] =
+                ((int*)listIndices[listId])[replaceOffset];
+        listIndicesReplaceOffset[vec] =
+                ((int*)listIndices[listId])[replaceOffset];
+    } else if (opt == INDICES_64_BIT) {
+        ((Index::idx_t*)listIndices[listId])[offset] =
+                ((Index::idx_t*)listIndices[listId])[replaceOffset];
+        listIndicesReplaceOffset[vec] =
+                ((Index::idx_t*)listIndices[listId])[replaceOffset];
+    }
+}
+
+void runIVFIndicesRemove(
+        Tensor<int, 1, true>& listIds,
+        Tensor<int, 1, true>& listOffset,
+        Tensor<int, 1, true>& listReplaceOffset,
+        Tensor<long, 1, true>& listIndicesReplaceOffset,
+        IndicesOptions opt,
+        DeviceVector<void*>& listIndices,
+        cudaStream_t stream) {
+    FAISS_ASSERT(
+            opt == INDICES_CPU || opt == INDICES_IVF || opt == INDICES_32_BIT ||
```

Comment:
this function should only allow `INDICES_32_BIT` and `INDICES_64_BIT`, as remove on `INDICES_IVF` doesn't make sense, and `INDICES_CPU` should be handled elsewhere (this file is just for actual GPU kernels)